specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json

```diff
@@ -2127,6 +2404,112 @@
+      "description": "The request body for a request to Bot Service Management to check availability of an Enterprise Channel name.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "the name of the Enterprise Channel for which availability needs to be checked.",
+          "type": "string"
+        }
+      }
+    },
+    "EnterpriseChannelCheckNameAvailabilityResponseBody": {
+      "description": "The response body returned for a request to Bot Service Management to check availability of an Enterprise Channel name.",
+      "type": "object",
+      "properties": {
+        "valid": {
+          "description": "indicates if the Enterprise Channel name is valid.",
+          "type": "boolean"
+        },
+        "message": {
+          "description": "additional message from the bot management api showing why a bot name is not available",
+          "type": "string"
+        }
+      }
+    },
+    "EnterpriseChannelResponseList": {
+      "properties": {
+        "nextLink": {
+          "description": "The link used to get the next page of bot service resources.",
+          "type": "string"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnterpriseChannel"
+          },
+          "description": "The list of Enterprise Channels and their properties."
+        }
+      },
+      "description": "The list of  bot service operation response."
+    },
+    "EnterpriseChannel": {
+      "type": "object",
+      "description": "Enterprise Channel resource definition",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EnterpriseChannelProperties",
+          "description": "The set of properties specific to an Enterprise Channel resource"
+        }
+      }
+    },
+    "EnterpriseChannelProperties": {
+      "properties": {
+        "state": {
+          "type": "string",
+          "description": "The current state of the Enterprise Channel"
+        },
+        "nodes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnterpriseChannelNode"
+          },
+          "description": "The nodes associated with the Enterprise Channel"
+        }
+      },
+      "description": "The parameters to provide for the Enterprise Channel.",
+      "required": [
+        "nodes"
+      ]
+    },
+    "EnterpriseChannelNode": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of Enterprise Channel Node"
+        },
+        "state": {
+          "type": "string",
+          "description": "The current state of the Enterprise Channel Node"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the Enterprise Channel Node"
+        },
+        "azureSku": {
```

Comment:
Do we need to declare an enum or a list api for these? What are the options?

Reply:
I can probably add this later. We don't have the options yet.